generated/nidcpower/nidcpower/lcr_measurement.py

```diff
@@ -0,0 +1,229 @@
+
+    _lcr_measurement_field_metadata = [
+        # field_name           label(s)
+        ("channel"           , "Channel"                                                             ),  # noqa: E202,E203
+        ("vdc"               , "DC voltage"                                                          ),  # noqa: E202,E203
+        ("idc"               , "DC current"                                                          ),  # noqa: E202,E203
+        ("stimulus_frequency", "Stimulus frequency"                                                  ),  # noqa: E202,E203
+        ("ac_voltage"        , "AC voltage"                                                          ),  # noqa: E202,E203
+        ("ac_current"        , "AC current"                                                          ),  # noqa: E202,E203
+        ("z"                 , "Impedance"                                                           ),  # noqa: E202,E203
+        ("series_lcr"        , ("Series inductance", "Series capacitance", "Series resistance")      ),  # noqa: E202,E203
+        ("parallel_lcr"      , ("Parallel inductance", "Parallel capacitance", "Parallel resistance")),  # noqa: E202,E203
+        ("d"                 , "Dissipation factor"                                                  ),  # noqa: E202,E203
+        ("measurement_mode"  , "Measurement mode"                                                    ),  # noqa: E202,E203
+        ("in_compliances"    , ("DC in compliance", "AC in compliance")                              ),  # noqa: E202,E203
+        ("unbalanced"        , "Unbalanced"                                                          ),  # noqa: E202,E203
+    ]
+
+    def __init__(self, data):
+        """LCRMeasurement
+
+        Creates and returns an LCRMeasurement object.
+
+        Args:
+            data (struct_NILCRMeasurement): The LCR measurement ctypes object returned by the driver.
+        """
+        self.channel = ""
+        self.vdc = data.vdc
+        self.idc = data.idc
+        self.stimulus_frequency = data.stimulus_frequency
+        self.ac_voltage = complex(data.ac_voltage_real, data.ac_voltage_imaginary)
+        self.ac_current = complex(data.ac_current_real, data.ac_current_imaginary)
+        self.z = complex(data.z_real, data.z_imaginary)
+        self.series_lcr = LCRMeasurement.LCR(
+            inductance=data.ls, capacitance=data.cs, resistance=data.rs
+        )
+        self.parallel_lcr = LCRMeasurement.LCR(
+            inductance=data.lp, capacitance=data.cp, resistance=data.rp
+        )
+        self.d = data.d
+        self.measurement_mode = enums.InstrumentMode(data.measurement_mode)
+        self.in_compliances = LCRMeasurement.InCompliances(
+            dc=bool(data.dc_in_compliance), ac=bool(data.ac_in_compliance)
+        )
+        self.unbalanced = bool(data.unbalanced)
+
+    @property
+    def z_magnitude_and_phase(self):
```

Comment:
Cstomers can get the impedance as complex and do the math themselves. I don't think there's much point in providing these utility properties.

Otherwise, why not have properties to return these as rectangular values?

Reply:
I'd rather err on the side of putting too little functionality than too much. See [wiki](https://github.com/ni/nimi-python/wiki/API-design-guidelines#deviate-from-c-apis-in-the-following-ways)

>If there's debate on whether a function should be included or not, then don't include it
Trivial to add if needed
Hard to remove once added

Reply:
These properties were specifically requested by Kunal:

"The helper functions also need to be part of the API as our specs are really tied to the 'derived measurements' not the complex versions."

> Otherwise, why not have properties to return these as rectangular values?

The `z` member itself is a rectangular value (it is stored as a complex number). `y` is created as a property that returns a rectangular value (complex number).

Reply:
I've spoken to Kunal and Sien. I understand the concern. I don't like to have redundant functionality but apparently portions of the user base cannot be trusted to know to extract the magnitude of a complex number.

However, there is redundant implementation here - the driver runtime already populates all these things, no need to recompute.

Consider chaning the implementation of this class so that there is minimal Python code getting executed and you rely more heavily in the values already returned by the driver runtime. That way there's less codepaths and less opportunities for bugs.